File: MercadoPagoSDKExamplesObjectiveC/MercadoPagoSDKExamplesObjectiveC/Hooks/ThirdHookViewController.m
//
//  ThirdHookViewController.m
//  MercadoPagoSDKExamplesObjectiveC
//
//  Created by Eden Torres on 11/27/17.
//  Copyright © 2017 MercadoPago. All rights reserved.
//

#import "ThirdHookViewController.h"

@interface ThirdHookViewController ()
@end

@implementation ThirdHookViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    NSLog(@"ThirdHookViewController loaded");
}

- (IBAction)didTapOnNext {
    if (self.actionHandler != nil) {
        [self.actionHandler next];
    }
}


#pragma mark - PXHookComponent mandatory delegates.
- (UIView * _Nonnull)render {
    return self.view;
}

- (enum PXHookStep)hookForStep {
    return PXHookStepBEFORE_PAYMENT;
}


#pragma mark - PXHookComponent optional delegates.
- (BOOL)shouldSkipHookWithHookStore:(PXHookStore * _Nonnull)hookStore {
    return NO;
}

- (void)renderDidFinish {
    NSLog(@"renderDidFinish");
}

- (BOOL)shouldShowBackArrow {
    return YES;
}

- (BOOL)shouldShowNavigationBar {
    return YES;
}

- (NSString * _Nullable)titleForNavigationBar {
    return @"Soy hook 3";
}

- (UIColor * _Nullable)colorForNavigationBar {
    return nil;
}

@end

File: PodTester/PodTester/Hooks/ThirdHookViewController.swift
//
//  ThirdHookViewController.swift
//  PodTester
//
//  Created by Juan sebastian Sanzone on 4/12/17.
//  Copyright © 2017 AUGUSTO COLLERONE ALFONSO. All rights reserved.
//

import UIKit
import MercadoPagoSDK

class ThirdHookViewController: UIViewController {

    fileprivate var navigationHandler: PXHookNavigationHandler?

    var targetHookStore: PXCheckoutStore?

    @IBOutlet weak var hookStoreDebugLabel: UILabel!

    override func viewDidLoad() {
        super.viewDidLoad()
    }

    @IBAction func didTapOnNext() {
        navigationHandler?.next()
    }
}

// MARK: - Hooks implementation.
extension ThirdHookViewController: PXHookComponent {

    func hookForStep() -> PXHookStep {
        return .BEFORE_PAYMENT
    }

    func render() -> UIView {
        return self.view
    }

    func titleForNavigationBar() -> String? {
        return "Hook 3"
    }

    func renderDidFinish() {
        if let savedValue = targetHookStore?.getData(forKey: "key"), let savedValueStr = savedValue as? String {
            hookStoreDebugLabel.text = savedValueStr.debugDescription
        } else {
            hookStoreDebugLabel.text = "{empty}"
        }
    }

    func didReceive(hookStore: PXCheckoutStore) {
        targetHookStore = hookStore
    }

    func shouldSkipHook(hookStore: PXCheckoutStore) -> Bool {
        if let paymentId = hookStore.getPaymentOptionSelected()?.getId(), paymentId == "bitcoin_payment" {
            return true
        }
        return false
    }

    func navigationHandlerForHook(navigationHandler: PXHookNavigationHandler) {
        self.navigationHandler = navigationHandler
    }
}

// MARK: - Presentation-Navigation
extension ThirdHookViewController {
    static func get() -> ThirdHookViewController {
        return HooksNavigationManager().getThirdHook()
    }
}
